client/src/partials/patients/edit/PatientGroupService.js

```diff
@@ -0,0 +1,77 @@
+angular.module('bhima.services')
+.service('PatientGroupModal', PatientGroupModal);
+
+PatientGroupModal.$inject = ['$uibModal'];
+
+/**
+ * This service is responsible for configuring and exposing modal that can be
+ * used to configure a patients groups subscription.
+ *
+ * @todo remove callbacks in favour of propegating values through promise fufillment
+ */
+function PatientGroupModal(Modal) {
+  var service = this;
+
+  var modalConfig = {
+    animation : true,
+    size : 'md',
+    keyboard : false,
+    backdrop : 'static'
+  };
+
+  service.updateDebtor = updateDebtor;
+  service.updateGroupConfig = updateGroupConfig;
+
+  /**
+   * This function if responsible for configuring and openening a modal to update
+   * debtor groups given a patient and a complete callback
+   *
+   * @params Object     patientEntity
+   */
+  function updateDebtor(patientEntity, callback) {
+    var debtorGroupConfig = {
+      templateUrl : 'partials/patients/edit/updateDebtorGroup.tmpl.html',
+      controller : 'UpdateDebtorGroup as UpdateDebtorGroupCtrl',
+      resolve : {
+        patient : function () {
+          return patientEntity;
+        },
+
+        /** @todo investigate passing multiple params through $modal.close to remove callback methods */
+        updateModel : function () {
```

Comment:
This is updating very old code (pre-2.X 2.X). It is also used to need to update the view without closing the modal for some reason, who knows why. 

It should work with Promises without a problem in the future.